### PR TITLE
[Feature][Connector-V2] Add mongodb connecter sink

### PR DESCRIPTION
**Pull request by wuchunfu**
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-mongodb-flink-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/mongodb/FakeSourceToMongodbIT.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        mongodbContainer.setPortBindings(Lists.newArrayList(String.format("%s:%s", MONGODB_PORT, MONGODB_PORT)));
+        Startables.deepStart(Stream.of(mongodbContainer)).join();
+        log.info("Mongodb container started");
+    }
+
+    @Test
+    public void testMongodbSource() throws IOException, InterruptedException {
+        Container.ExecResult execResult = executeSeaTunnelFlinkJob("/mongodb/fake_to_mongodb.conf");
+        Assertions.assertEquals(0, execResult.getExitCode());
+    }
```
Comment (CalvinKirs) [inline seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-mongodb-flink-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/mongodb/FakeSourceToMongodbIT.java]: Could you verify that the sink data is written as you expected?

Comment (wuchunfu) [reply seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-mongodb-flink-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/mongodb/FakeSourceToMongodbIT.java]: > Could you verify that the sink data is written as you expected?

I have verified locally, but I don't know how to verify with the program, can you provide some help?

Comment (CalvinKirs) [reply seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-mongodb-flink-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/mongodb/FakeSourceToMongodbIT.java]: You can refer the IOTDB Sink e2e

**Comment by hailin0** [comment]
add spark e2e-testcase?

**Comment by wuchunfu** [comment]
> add spark e2e-testcase?

OK, thank you for your reminder. I'll add it later

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-mongodb-flink-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/mongodb/FakeSourceToMongodbIT.java (hunk 1 above)
Comment (wuchunfu) [reply seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-mongodb-flink-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/mongodb/FakeSourceToMongodbIT.java]: > You can refer the IOTDB Sink e2e

OK, Thanks

Comment (wuchunfu) [reply seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-mongodb-flink-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/mongodb/FakeSourceToMongodbIT.java:74]: > You can refer the IOTDB Sink e2e

OK, Thanks

File: docs/en/connector-v2/sink/MongoDB.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+> MongoDB sink connector
+
+## Description
+
+Write data to `MongoDB`
+
+## Key features
+
+- [x] [batch](../../concept/connector-v2-features.md)
+- [ ] [stream](../../concept/connector-v2-features.md)
```
Comment (Hisoka-X) [inline docs/en/connector-v2/sink/MongoDB.md]: In ours seatunnel sink api model, always support batch and stream

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-mongodb-flink-e2e/src/test/resources/mongodb/fake_to_mongodb.conf [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+source {
+  # This is a example source plugin **only for test and demonstrate the feature source plugin**
+    FakeSource {
+      result_table_name = "fake"
+      schema = {
+        fields {
+          id = "int"
+          name = "string"
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-mongodb-flink-e2e/src/test/resources/mongodb/fake_to_mongodb.conf]: test all datatypes?

File: docs/en/connector-v2/sink/MongoDB.md (hunk 2 above)
Comment (wuchunfu) [reply docs/en/connector-v2/sink/MongoDB.md]: > In ours seatunnel sink api model, always support batch and stream

Thanks for the reminder, I'll fix it

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-mongodb-flink-e2e/src/test/resources/mongodb/fake_to_mongodb.conf (hunk 3 above)
Comment (wuchunfu) [reply seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-mongodb-flink-e2e/src/test/resources/mongodb/fake_to_mongodb.conf]: > test all datatypes?

Thanks for the reminder, I'll fix it

File: seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/sink/MongodbSinkWriter.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    private final SerializationSchema serializationSchema;
+
+    private MongoClient client;
+
+    public MongodbSinkWriter(SeaTunnelRowType rowType, MongodbParameters params) {
+        this.rowType = rowType;
+        this.params = params;
+        // TODO according to format to initialize serializationSchema
+        // Now temporary using json serializationSchema
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/sink/MongodbSinkWriter.java]: MongoDB only support insert json, so this tips is useless.

File: seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/sink/MongodbSinkWriter.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        this.serializationSchema = new JsonSerializationSchema(rowType);
+    }
+
+    @Override
+    public void write(SeaTunnelRow rows) throws IOException {
+        byte[] serialize = serializationSchema.serialize(rows);
+        String content = new String(serialize);
+
+        String database = this.params.getDatabase();
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/sink/MongodbSinkWriter.java]: How about using these variables include `database` `collection` `client` as class attributes? These attributes can be initialized in constructor method.

File: seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/sink/MongodbSinkWriter.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            .getDatabase(database)
+            .getCollection(collection);
+
+        Document doc = Document.parse(content);
+        mongoCollection.insertOne(doc);
+    }
+
+    @Override
+    public void close() throws IOException {
+        if (client != null) {
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/sink/MongodbSinkWriter.java:68]: Please double check mongoCollection whether need be closed.

Comment (wuchunfu) [reply seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/sink/MongodbSinkWriter.java:68]: > Please double check mongoCollection whether need be closed.

mongoCollection has no close method

File: seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/sink/MongodbSinkWriter.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        this.collection = this.params.getCollection();
+        this.serializationSchema = new JsonSerializationSchema(rowType);
+    }
+
+    @Override
+    public void write(SeaTunnelRow rows) throws IOException {
+        byte[] serialize = serializationSchema.serialize(rows);
+        String content = new String(serialize);
+
+        this.client = MongoClients.create(params.getUri());
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/sink/MongodbSinkWriter.java]: Why not use these two variables as class attributes, too? Each time a piece of data is written, a client object and a collection object are retrieved, this results in unnecessary method calls and resource consumption.

Comment (wuchunfu) [reply seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/sink/MongodbSinkWriter.java]: > Why not use these two variables as class attributes, too? Each time a piece of data is written, a client object and a collection object are retrieved, this results in unnecessary method calls and resource consumption.

Makes sense, I was negligent, thanks for the reminder

File: seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/sink/MongodbSinkWriter.java [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        this.serializationSchema = new JsonSerializationSchema(rowType);
+    }
+
+    @Override
+    public void write(SeaTunnelRow rows) throws IOException {
+        byte[] serialize = serializationSchema.serialize(rows);
+        String content = new String(serialize);
+
+        MongoCollection<Document> mongoCollection = this.client
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/sink/MongodbSinkWriter.java]: `MongoCollection<Document> mongoCollection` can be a class attribute too.

**Review by TyrantLucifer — APPROVED** [review]
LGTM.

File: seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/sink/MongodbSinkWriter.java (hunk 8 above)
Comment (wuchunfu) [reply seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/sink/MongodbSinkWriter.java]: Makes sense, done

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Pull request merged**